### PR TITLE
Use curl to upload packages rather than pkg upload

### PR DESCRIPTION
**Pull request by scotthain**
Signed-off-by: Scott Hain <shain@chef.io>

this is a temporary fix until https://github.com/habitat-sh/builder/issues/940 gets resolved (which is underway)

### CODE REVIEW

File: .buildkite/scripts/build_component.sh [hunk 1]
```diff
@@ -73,7 +73,15 @@ esac
...
+# ${hab_binary} pkg upload \
+#     --channel="${channel}" \
+#     --auth="${HAB_AUTH_TOKEN}" \
+#     "results/${pkg_artifact}"
+
+curl -v -X POST \
+    -H "Accept: application/json" \
+    -H "Content-Type: application/json" \
+    -H "Authorization: Bearer $HAB_AUTH_TOKEN" \
+    --data-binary "@/hab/cache/artifacts/${pkg_artifact}.hart" "http://bldr.habitat.sh:9636/v1/depot/pkgs/${pkg_ident}\?checksum=${pkg_sha256sum}\?target=${pkg_target}"
```
Comment (christophermaier) [inline .buildkite/scripts/build_component.sh]: I think it needs to be `--data-binary "*results/${pkg_artifact}"`.

It'd also be good to put the URL down on the next line by itself.

Comment (christophermaier) [reply .buildkite/scripts/build_component.sh]: Also, you've got two `?` in the URL.

Comment (scotthain) [reply .buildkite/scripts/build_component.sh]: Done!

File: .buildkite/scripts/build_component.sh [hunk 2]
```diff
@@ -73,7 +73,16 @@ esac
...
+#     --channel="${channel}" \
+#     --auth="${HAB_AUTH_TOKEN}" \
+#     "results/${pkg_artifact}"
+
+curl -v -X POST \
+    -H "Accept: application/json" \
+    -H "Content-Type: application/json" \
+    -H "Authorization: Bearer $HAB_AUTH_TOKEN" \
+    --data-binary "@results/${pkg_artifact}.hart" \
+    "http://bldr.habitat.sh:9636/v1/depot/pkgs/${pkg_ident}\?checksum=${pkg_sha256sum:?}\&target=${pkg_target}"
```
Comment (chefsalim) [inline .buildkite/scripts/build_component.sh]: Hmm checksum may need to be blake2b as per 
```
pub fn checksum(&self) -> Result<String> { hash::hash_file(&self.path) } 
```
unless I'm reading the archive code wrong ?

Comment (raskchanky) [reply .buildkite/scripts/build_component.sh]: @chefsalim is correct.  It's the same hash you get from `hab pkg hash`

File: .buildkite/scripts/build_component.sh [hunk 3]
```diff
@@ -73,7 +73,16 @@ esac
...
+#     --channel="${channel}" \
+#     --auth="${HAB_AUTH_TOKEN}" \
+#     "results/${pkg_artifact}"
+
+curl -v -X POST \
+    -H "Accept: application/json" \
+    -H "Content-Type: application/json" \
+    -H "Authorization: Bearer $HAB_AUTH_TOKEN" \
+    --data-binary "@results/${pkg_artifact}.hart" \
+    "http://bldr.habitat.sh:9636/v1/depot/pkgs/${pkg_ident}\?checksum=${pkg_blake2bsum:?}\&target=${pkg_target}"
```
Comment (chefsalim) [inline .buildkite/scripts/build_component.sh]: So we also need to do a promote to the channel specified..

Comment (chefsalim) [reply .buildkite/scripts/build_component.sh]: You can just do a 'hab pkg promote...'

Comment (baumanj) [reply .buildkite/scripts/build_component.sh]: I don't think you want those escapes inside a quoted string
```suggestion
    "http://bldr.habitat.sh:9636/v1/depot/pkgs/${pkg_ident}?checksum=${pkg_blake2bsum:?}&target=${pkg_target}"
```

**Pull request merged**